miv/io/data.py

```diff
@@ -185,6 +188,153 @@ def set_channel_mask(self, channel_id: Iterable[int]):
+                )
+            )
+
+        mean = np.mean(dot_products)
+        threshold = mean + statistics.stdev(dot_products)
+
+        mask_list = []
+        for chan in range(num_channels):
+            if dot_products[chan] > threshold:
+                mask_list.append(chan)
+        self.add_channel_mask(np.concatenate((mask_list, exp_binned[2])))
+
+    def get_binned_matrix(
+        self,
+        filter: FilterProtocol,
+        detector: SpikeDetectionProtocol,
+        offset: float = 0,
+        bins_per_second: float = 100,
+    ):
+        """
+        Performs spike detection and return a binned 2D matrix with columns being the
+        binned number of spikes from each channel.
+
+        Parameters
+        ----------
+        filter : FilterProtocol
+            Filter that is applied to the signal before masking.
+        detector : SpikeDetectionProtocol
+            Spike detector that extracts spikes from the signals.
+        offset : float
+            The time in seconds to be trimmed in front (default = 0).
+        bins_per_second : float
+            Optional parameter for binning spikes with respect to time.
+            The spikes are binned for comparison between the spontaneous recording and
+            the other experiments. This value should be adjusted based on the firing rate.
+            A high value reduces type I error; a low value reduces type II error.
+            As long as this value is within a reasonable range, it should negligibly affect
+            the result (see jupyter notebook demo).
+
+        Returns
+        -------
+        [binned_matrix, num_bins, empty_channels]
+        *binned_matrix* has columns as channels
+        *empty_channels* is an array of indices of empty channels
+        """
+
+        result = []
+        with self.load() as (sig, times, samp):
+            start_time = times[0] + offset
+            starting_index = int(offset * samp)
+
+            trimmed_signal = sig[starting_index:]
+            trimmed_times = times[starting_index:]
+
+            filtered_sig = filter(trimmed_signal, samp)
+            spiketrains = detector(filtered_sig, trimmed_times, samp)
+
+            bins_array = np.arange(
+                start=start_time, stop=trimmed_times[-1], step=1 / bins_per_second
+            )
+            num_bins = len(bins_array)
+            num_channels = len(spiketrains)
+            empty_channels = []
+
+            for chan in range(num_channels):
+                if len(spiketrains[chan]) == 0:
+                    empty_channels.append(chan)
+
+                spike_counts = np.zeros(shape=num_bins + 1, dtype=int)
+                digitized_indices = np.digitize(spiketrains[chan], bins_array)
+                for bin_index in digitized_indices:
+                    spike_counts[bin_index] += 1
+                result.append(spike_counts)
+
+        return [np.transpose(result), num_bins, empty_channels]
```

Comment:
Why return as list? Can we just do `return np.transpose(result), num_bins, empty_channels`?